controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -218,35 +220,16 @@ func (r *AutoscalingRunnerSetReconciler) Reconcile(ctx context.Context, req ctrl
 		log.Info("Find existing ephemeral runner set", "name", runnerSet.Name, "specHash", runnerSet.Labels[labelKeyRunnerSpecHash])
 	}
 
-	if desiredSpecHash != latestRunnerSet.Labels[labelKeyRunnerSpecHash] {
-		log.Info("Latest runner set spec hash does not match the current autoscaling runner set. Creating a new runner set")
-		return r.createEphemeralRunnerSet(ctx, autoscalingRunnerSet, log)
-	}
-
-	oldRunnerSets := existingRunnerSets.old()
-	if len(oldRunnerSets) > 0 {
-		log.Info("Cleanup old ephemeral runner sets", "count", len(oldRunnerSets))
-		err := r.deleteEphemeralRunnerSets(ctx, oldRunnerSets, log)
-		if err != nil {
-			log.Error(err, "Failed to clean up old runner sets")
-			return ctrl.Result{}, err
-		}
-	}
-
 	// Make sure the AutoscalingListener is up and running in the controller namespace
 	listener := new(v1alpha1.AutoscalingListener)
 	if err := r.Get(ctx, client.ObjectKey{Namespace: r.ControllerNamespace, Name: scaleSetListenerName(autoscalingRunnerSet)}, listener); err != nil {
 		if kerrors.IsNotFound(err) {
```

Comment:
If we fail to fetch listener, we should return with an error. It is likely that the API call failed, and we need to reconcile again

Reply:
Fixed